ocean/src/clans.rs

```diff
@@ -1,39 +1,57 @@
+use std::collections::{HashMap, HashSet};
 
 #[derive(Debug)]
 pub struct ClanSystem {
     // TODO: add necessary fields
+    clans: HashMap<String, HashSet<String>>, 
 }
 
 impl ClanSystem {
     pub fn new() -> ClanSystem {
-        unimplemented!();
+        ClanSystem {clans: HashMap::new()}
     }
 
     /**
      * Returns a list of the names of the clan members for the given clan id.
      */
     pub fn get_clan_member_names(&self, clan_id: &str) -> Vec<String> {
-        unimplemented!();
+        //unimplemented!();
+        if let Some(members) = self.clans.get(clan_id) {
+            members.iter().cloned().collect()
+        } else {
+            Vec::new()
+        }
     }
 
     /**
      * Returns the number of clans currently in existence.
      */
     pub fn get_clan_count(&self) -> usize {
-        unimplemented!();
+        //unimplemented!();
+        self.clans.len()
     }
 
     /**
      * Returns the number of clan members for the given clan id.
      */
     pub fn get_clan_member_count(&self, clan_id: &str) -> usize {
-        unimplemented!();
+        //unimplemented!();
+        if let Some(members) = self.clans.get(clan_id) {
+            members.len()
+        } else {
```

Comment:
Your code looks fine, but here are a small improvement and suggestion:
Instead of pattern matching on Option, you can use the get method to directly access the value associated with the given key like this:
`self.clans.get(clan_id).map_or(0, |members| members.len())`